examples/virtio-snd/snd_driver_vmm.c

```diff
@@ -0,0 +1,252 @@
+        return;
+    }
+    
+    /* Initialise the virtual GIC driver */
+    bool success = virq_controller_init(GUEST_VCPU_ID);
+    if (!success) {
+        LOG_VMM_ERR("Failed to initialise emulated interrupt controller\n");
+        return;
+    }
+
+    assert(serial_rx_data);
+    assert(serial_tx_data);
+    assert(serial_rx_active);
+    assert(serial_rx_free);
+    assert(serial_tx_active);
+    assert(serial_tx_free);
+
+    
+    /* Initialise our sDDF ring buffers for the serial device */
+    serial_queue_handle_t rxq, txq;
+    serial_queue_init(&rxq,
+                      (serial_queue_t *)serial_rx_free,
+                      (serial_queue_t *)serial_rx_active,
+                      true,
+                      NUM_ENTRIES,
+                      NUM_ENTRIES);
+    for (int i = 0; i < NUM_ENTRIES - 1; i++) {
+        int ret = serial_enqueue_free(&rxq,
+                               serial_rx_data + (i * BUFFER_SIZE),
+                               BUFFER_SIZE);
+        if (ret != 0) {
+            microkit_dbg_puts(microkit_name);
+            microkit_dbg_puts(": server rx buffer population, unable to enqueue buffer\n");
+        }
+    }
+    serial_queue_init(&txq,
+                      (serial_queue_t *)serial_tx_free,
+                      (serial_queue_t *)serial_tx_active,
+                      true,
+                      NUM_ENTRIES,
+                      NUM_ENTRIES);
+    for (int i = 0; i < NUM_ENTRIES - 1; i++) {
+        // Have to start at the memory region left of by the rx ring
+        int ret = serial_enqueue_free(&txq,
+                               serial_tx_data + ((i + NUM_ENTRIES) * BUFFER_SIZE),
+                               BUFFER_SIZE);
+        assert(ret == 0);
+        if (ret != 0) {
+            microkit_dbg_puts(microkit_name);
+            microkit_dbg_puts(": server tx buffer population, unable to enqueue buffer\n");
+        }
+    }
+
+    /* Initialise virtIO console device */
+    success = virtio_mmio_console_init(&virtio_console,
+                                  VIRTIO_CONSOLE_BASE,
+                                  VIRTIO_CONSOLE_SIZE,
+                                  VIRTIO_CONSOLE_IRQ,
+                                  &rxq, &txq,
+                                  SERIAL_MUX_TX_CH);
+    assert(success);
+    
+    success = virq_register(GUEST_VCPU_ID, UIO_SND_IRQ, &uio_sound_virq_ack, NULL);
+    assert(success);
+
+    success = fault_register_vm_exception_handler(UIO_SND_FAULT_ADDRESS,
+                                                  sizeof(size_t),
+                                                  &uio_sound_fault_handler, NULL);
+    assert(success);
+
+#if defined(BOARD_qemu_arm_virt)
+    register_passthrough_irq(37, 5);
+#elif defined(BOARD_odroidc4)
+    register_passthrough_irq(48, 5);
+    register_passthrough_irq(63, 6);
+    register_passthrough_irq(62, 7);
+    register_passthrough_irq(5, 8);
+#else
```

Comment:
These IRQ numbers should all have names, I shouldn't have to refer to the DTS to figure out what they are.

Reply:
I labelled them with comments in the most recent commit. Not sure if its worth putting # defines since the amount of interrupt types changes per device class.